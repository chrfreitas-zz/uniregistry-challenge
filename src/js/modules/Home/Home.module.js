import React, { Component } from 'react';

import DomainsTable from '../../components/DomainsTable/DomainsTable.component';
import Modal from '../../components/Modal/Modal.component';

class Home extends Component {
    state = {
        domains: [
            {
                id: 1,
                name: 'foolfighter.lol',
                uniregistry: 'yes',
                price: '$12.00'
            },
            {
                id: 2,
                name: 'selfdriving.cars',
                uniregistry: 'yes',
                price: '$16.00'
            },
            {
                id: 3,
                name: 'greendiamondsky.com',
                uniregistry: 'no',
                price: '$9.00'
            },
        ],
        modal: {
            opened: false,
            domain: {}
        }
    }

    update = (domain) => {
        const newState = {
            domains: this.state.domains.map(elem => {
                if(elem.id === domain.id){
                    elem = domain;
                }

                return elem;
            })
        }

        this.setState(newState);
    }

    toggleModal = (domain) => {
        this.setState(prevState => {
            return {
                modal: {
                    opened: !prevState.modal.opened,
                    domain: prevState.domains.find(item => item.id === domain.id)
                }
            }
        });
    }

    render() {
        return (
            <div>
                <DomainsTable domains={this.state.domains} toggleModal={this.toggleModal}/>
                { this.state.modal.opened && <Modal domain={this.state.modal.domain} update={this.update}/> }
            </div>
        )
    }
}


export default Home;