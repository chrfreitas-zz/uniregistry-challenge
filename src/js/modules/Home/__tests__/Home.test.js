import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Home from '../Home.module';

beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });

    class LocalStorageMock {
        constructor() {
            this.store = {};
        }

        clear() {
            this.store = {};
        }

        getItem(key) {
            return this.store[key] || null;
        }

        setItem(key, value) {
            this.store[key] = value.toString();
        }

        removeItem(key) {
            delete this.store[key];
        }
    };

    global.localStorage = new LocalStorageMock;

    global.fetch = (url) => new Promise((resolve, reject) => {
        resolve({
            name: 1
        })
    });
});

describe('Home Module', () => {
    it('should match with the snapshot', () => {
        const wrapper = renderer.create(<Home />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('should change set editMode true', () => {
        const domain = {
            id: 2,
            name: 'lalala',
            email: 'lalala@gmail.com',
            price: '$3.00'
        }

        const wrapper = shallow(<Home />);
        wrapper.instance().goEdit(domain);
        expect(wrapper.state().editMode).toBeTruthy();
    })

    it('should update the domain object with the new object', () => {
        // const domain = {
        //     id: 2,
        //     name: 'lalala',
        //     email: 'lalala@gmail.com',
        //     price: '$3.00'
        // }

        // const wrapper = shallow(<Home />);
        // wrapper.instance().update(domain);

        // const newDomain = wrapper.state().domains.find(item => item.id === domain.id);

        // expect(newPrice.price).toBe(domain.price);
    })

    it('should show edit mode when true', () => {
        const wrapper = shallow(<Home />);
        wrapper.instance().setEditMode(true);

        expect(wrapper.state().editMode).toBeTruthy();
    });

    it('should hide edit mode when false', () => {
        const wrapper = shallow(<Home />);
        wrapper.instance().setEditMode(false);

        expect(wrapper.state().editMode).toBeFalsy();
    });
});
