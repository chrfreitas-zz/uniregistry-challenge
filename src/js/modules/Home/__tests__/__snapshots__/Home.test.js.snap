// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Module should match with the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="row justify-content-center"
      >
        <div
          className="col col-md-10 col-lg-6"
        >
          <div
            className="domain-table"
          >
            <table
              className="table table-hover"
            >
              <thead>
                <tr>
                  <th>
                    Domains name
                  </th>
                  <th>
                    Uniregistry
                  </th>
                  <th>
                    Price
                  </th>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>
          <div
            className="domain-form"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
